cql3/statements/select_statement.cc

```diff
@@ -1398,12 +1399,25 @@ indexed_table_select_statement::find_index_partition_ranges(query_processor& qp,
     using value_type = std::tuple<dht::partition_range_vector, lw_shared_ptr<const service::pager::paging_state>>;
     auto now = gc_clock::now();
     auto timeout = db::timeout_clock::now() + get_timeout(state.get_client_state(), options);
+    bool is_paged = options.get_page_size() >= 0;
+    constexpr size_t MAX_PARTITION_RANGES = 1000;
+    // Check that `partition_range_vector` won't cause a large allocation (see #18536).
+    // If this warning appears, please adjust MAX_PARTITION_RANGES accordingly.
+    if (MAX_PARTITION_RANGES * sizeof(dht::partition_range_vector::value_type) > current_allocator().preferred_max_contiguous_allocation()) {
```

Comment:
You can code 128k here and convert it to static_assert, there's no way we'd use anything other than the standard allocator.

Yes, we should have a named constant for it, but we don't. At this point 128k is like Plank's constant for us.